test.ts

```diff
@@ -480,6 +480,52 @@ describe('Project', function () {
     );
   });
 
+  it('adjusts peerDependencies of linked dependencies', function () {
+    let baseProject = new Project('base');
+    let alpha = baseProject.addDependency('alpha', {
+      files: {
+        'index.js': `
+          exports.betaVersion = function() {
+            return require('beta/package.json').version;
+          }
+          exports.gammaLocation = function() {
+            return require.resolve('gamma/package.json');
+          }
+        `,
+        deeper: {
+          'index.js': `
+            exports.betaVersion = function() {
+              return 'inner' + require('beta/package.json').version;
+            }
+          `,
+        },
+      },
+    });
+    alpha.pkg.peerDependencies = { beta: '^1.0.0' };
+    alpha.addDependency('gamma');
+    baseProject.addDependency('beta', { version: '1.1.0' });
+    baseProject.writeSync();
+
+    // precondition: in the baseProject, alpha sees its beta peerDep as beta@1.1.0
+    expect(require(require.resolve('alpha', { paths: [baseProject.baseDir] })).betaVersion()).to.eql('1.1.0');
```

Comment:
`require` caches so for a test I believe we should treat these files sources as live.

Either by purging the cache prior, or if possible using `fs.readJsonSync`

Reply:
Fixed.

Reply:
it still looks like we are using `require` here, how was this fixed? I might be missing something obvious...